### PR TITLE
Fix container restart on every run when image has health checks

### LINKED ISSUE #1180: Containers with embedded health checks restart on every Chef run
## :ghost: Brief Description

After https://github.com/sous-chefs/docker/issues/1130, for `docker_container` resources that embed an health check in their Dockerfile, get restarted on every chef run.

## :pancakes: Cookbook version

8.2.2

## :woman_cook:  Chef-Infra Version

16.6.14

## :tophat: Platform details

Ubuntu 20.04

## Steps To Reproduce

```
docker_container 'cadvisor' do
  repo 'cadvisor'
  tag 'latest'
end
```

## :police_car: Expected behavior

Multiple runs should not restart container every time

## :heavy_plus_sign: Additional context

```
[dev-eng] out:     * docker_container[cadvisor] action run
[dev-eng] out:       - stopping cadvisor (will kill after 10s)
[dev-eng] out:       - deleting cadvisor
[dev-eng] out:       - update cadvisor
[dev-eng] out:       -   set health_check to {} (was {"Test"=>["CMD-SHELL", "curl -f http://localhost:8080/healthz || exit 1"], "Interval"=>30000000000, "Timeout"=>3000000000})
[dev-eng] out:       - starting cadvisor
```

very similar to https://github.com/sous-chefs/docker/issues/1139

### PR DESCRIPTION
**Pull request by gionn**
# Description

Fix #1180

## Check List

- [ ] All tests pass. See TESTING.md for details.
- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable.

### CODE REVIEW

File: libraries/docker_container.rb [hunk 1]
```diff
@@ -28,7 +28,7 @@ class DockerContainer < DockerBase
     property :extra_hosts, [Array, nil], coerce: proc { |v| Array(v).empty? ? nil : Array(v) }
     property :exposed_ports, PartialHashType, default: {}
     property :force, [true, false], default: false, desired_state: false
-    property :health_check, Hash, default: {}
+    property :health_check, [Hash, nil]
```
Comment (damacus) [inline libraries/docker_container.rb:31]:
```suggestion
    property :health_check, Hash
```

I'm fairly sure this is the same thing.

**Review by damacus — COMMENTED** [review]
Just a little style thing but this is basically ready to go 👍🏼

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,11 +7,13 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
 ## Unreleased
 
+- Ensure docker_container :health_check is idempotent
+
 ## 8.2.3 - *2021-09-08*
 
 - Fix private registries credentials handling and public registries
 
-## 8.2.2 - *2021-08-27*
+# 8.2.2 - *2021-08-27*
```
Comment (gionn) [inline CHANGELOG.md:16]:
```suggestion
## 8.2.2 - *2021-08-27*
```

**Review by ramereth — APPROVED** [review]

**Comment by kitchen-porter** [comment]
Released as: [8.2.4](https://github.com/sous-chefs/docker/releases/tag/8.2.4)

**Comment by gionn** [comment]
Thanks to all!

**Pull request merged**